moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -303,36 +346,100 @@ moveit::core::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const
 
       if (default_solver_plugin_.empty())
       {
-        RCLCPP_DEBUG(LOGGER, "Loading settings for kinematics solvers from the ROS param server ...");
+        // construct a parameter client to get parameters from the move_group node
+        using namespace std::chrono_literals;
+        auto parameters_client = std::make_shared<rclcpp::AsyncParametersClient>(node_, "move_group");
+        while (!parameters_client->wait_for_service(1s)) {
```

Comment:
I'm not a fan of this kind of busy waiting and shutting down on failure, but I see that the current API doesn't really provide other options. @tylerjw here is another use case for improved error handling ;). For now, I would definitely return the function after calling shutdown.

Reply:
Yeah, ideally this would just defer loading the plugins until the parameters could be found, but all of the control flow up to this point is synchronous and would likely require a much larger re-architecture effort.